v2/tools/asoctl/internal/importing/importable_arm_resource.go

```diff
@@ -143,22 +155,30 @@ func (i *importableARMResource) createImportFunction(
 	// Loader is a function that does the actual loading
 	loader := i.loader()
 
-	// Check to see if we have an extension to customize loading, and if we do, wrap the loader
+	// Check to see if we have an extension to customize loading, and if we do, wrap importFn
 	gvk := instance.GetObjectKind().GroupVersionKind()
 	if ex, ok := i.GetResourceExtension(gvk); ok {
 		next := loader
-		loader = func(ctx context.Context, resource genruntime.ImportableResource) (extensions.ImportResult, error) {
-			return ex.Import(ctx, resource, next)
+		loader = func(
+			ctx context.Context,
+			resource genruntime.ImportableResource,
+			owner *genruntime.ResourceReference,
+		) (extensions.ImportResult, error) {
+			return ex.Import(ctx, resource, owner, next)
 		}
 	}
 
+	// Lastly, we need to wrap importFn to remove Status
```

Comment:
Explain why status needs to be cleared here, rather than (or in addition to) that it must be cleared.

I assume the reason is because if status is not cleared, it will show up in the exported resource YAML which is both confusing and not-helpful?

Reply:
Correct. I'll update the code.